types/meteor/check.d.ts

```diff
@@ -1,45 +1,84 @@
-declare module "meteor/check" {
+declare module 'meteor/check' {
+    /**
+     * The namespace for all Match types and methods.
+     */
     module Match {
-        interface Matcher<T> { _meteorCheckMatcherBrand: void }
+        interface Matcher<T> {
+            _meteorCheckMatcherBrand: void;
+        }
         export type Pattern =
-            typeof String |
-            typeof Number |
-            typeof Boolean |
-            typeof Object |
-            typeof Function |
-            (new (...args: any[]) => any) |
-            undefined | null | string | number | boolean |
-            [Pattern] |
-            {[key: string]: Pattern} |
-            Matcher<any>;
-        export type PatternMatch<T extends Pattern> =
-            T extends Matcher<infer U> ? U :
-            T extends typeof String ? string :
-            T extends typeof Number ? number :
-            T extends typeof Boolean ? boolean :
-            T extends typeof Object ? object :
-            T extends typeof Function ? Function :
-            T extends undefined | null | string | number | boolean ? T :
-            T extends new (...args: any[]) => infer U ? U :
-            T extends [Pattern] ? PatternMatch<T[0]>[] :
-            T extends {[key: string]: Pattern} ? {[K in keyof T]: PatternMatch<T[K]>} :
-            unknown;
+            | typeof String
+            | typeof Number
+            | typeof Boolean
+            | typeof Object
+            | typeof Function
+            | (new (...args: any[]) => any)
+            | undefined
+            | null
+            | string
+            | number
+            | boolean
+            | [Pattern]
+            | { [key: string]: Pattern }
+            | Matcher<any>;
+        export type PatternMatch<T extends Pattern> = T extends Matcher<infer U>
+            ? U
+            : T extends typeof String
+            ? string
+            : T extends typeof Number
+            ? number
+            : T extends typeof Boolean
+            ? boolean
+            : T extends typeof Object
+            ? object
+            : T extends typeof Function
+            ? Function
+            : T extends undefined | null | string | number | boolean
+            ? T
+            : T extends new (...args: any[]) => infer U
+            ? U
+            : T extends [Pattern]
+            ? PatternMatch<T[0]>[]
+            : T extends { [key: string]: Pattern }
+            ? { [K in keyof T]: PatternMatch<T[K]> }
+            : unknown;
```

Comment:
In my subjective opinion, the previous version was much better. I know it doesn't follow any strict formatting rules (I guess you've just used ESLint, Prettier, or both), but in this very case, I'd keep these in a one-line-per-branch manner.